src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/BuyMine.java

```diff
@@ -1,34 +1,64 @@
+import com.github.splendor_mobile_game.game.model.Card;
+import com.github.splendor_mobile_game.game.model.Game;
+import com.github.splendor_mobile_game.game.model.Room;
+import com.github.splendor_mobile_game.game.model.User;
+import com.github.splendor_mobile_game.websocket.communication.ServerMessage;
 import com.github.splendor_mobile_game.websocket.communication.UserMessage;
+import com.github.splendor_mobile_game.websocket.handlers.DataClass;
 import com.github.splendor_mobile_game.websocket.handlers.Messenger;
 import com.github.splendor_mobile_game.websocket.handlers.Reaction;
 import com.github.splendor_mobile_game.websocket.handlers.ReactionName;
+import com.github.splendor_mobile_game.websocket.handlers.ServerMessageType;
+import com.github.splendor_mobile_game.websocket.handlers.exceptions.CardDoesntExistException;
+import com.github.splendor_mobile_game.websocket.handlers.exceptions.RoomDoesntExistException;
+import com.github.splendor_mobile_game.websocket.handlers.exceptions.UserAlreadyInRoomException;
+import com.github.splendor_mobile_game.websocket.handlers.exceptions.UserNotAMemberException;
+import com.github.splendor_mobile_game.websocket.handlers.exceptions.UserTurnException;
+import com.github.splendor_mobile_game.websocket.response.ErrorResponse;
+import com.github.splendor_mobile_game.websocket.response.Result;
+import com.github.splendor_mobile_game.websocket.utils.Log;
 
 /**
  * Players send this request when it is their turn and they want to buy a mine card that is on the table.
  * The server responds with a message of type `BUY_MINE_ANNOUNCEMENT` to all players, announcing that the purchase has been made.
  * 
  * Example of user request:
  * {
- *      "messageContextId": "02442d1b-2095-4aaa-9db1-0dae99d88e03",
+ *      "contextId": "02442d1b-2095-4aaa-9db1-0dae99d88e03",
  *      "type": "BUY_MINE",
  *      "data": {
- *          "userUuid": "6850e6c1-6f1d-48c6-a412-52b39225ded7",
- *          "cardUuid": "521ba578-f989-4488-b3ee-91b043abbc83"
+ *          "userDTO":{
+ *              "uuid": "6850e6c1-6f1d-48c6-a412-52b39225ded7"
+ *          },
+ *          "cardDTO":{
+ *              "uuid": "521ba578-f989-4488-b3ee-91b043abbc83"
+ *          }
  *      }
  * }
  * 
  * Example of server announcement:
  * {
- *      "messageContextId": "02442d1b-2095-4aaa-9db1-0dae99d88e03",
+ *      "contextId": "02442d1b-2095-4aaa-9db1-0dae99d88e03",
  *      "type": "BUY_MINE_ANNOUNCEMENT",
  *      "result": "OK",
  *      "data": {
- *          "userUuid": "6850e6c1-6f1d-48c6-a412-52b39225ded7",
- *          "userName": "John",
- *          "cardUuid": "521ba578-f989-4488-b3ee-91b043abbc83"
+ *          "buyerUser":{
+ *              "uuid":6850e6c1-6f1d-48c6-a412-52b39225ded7
+ *          },
+ *          "boughtCard":{
+ *              "uuid": "521ba578-f989-4488-b3ee-91b043abbc83"
+ *          },
+ *          "newCardRevealed":{
+ *              "uuid": 501ba578-f919-4488-b3ee-91b043abbc83
```

Comment:
Just UUID of the new card is not enough. When the game starts, players only get the revealed cards and info about then such as the cost and what bonus token it provides. So there we also have to send full information about this card, not only uuid.